app/services/PayeAtsService.scala

```diff
@@ -28,15 +31,20 @@ import scala.concurrent.Future
 
 object PayeAtsService extends PayeAtsService{
   override val middleConnector = MiddleConnector
+  override val auditService = AuditService
 }
 
 trait PayeAtsService {
   def middleConnector: MiddleConnector
+  def auditService: AuditService
 
-  def getPayeATSData(nino: Nino, taxYear: Int)(implicit hc: HeaderCarrier):Future[Either[HttpResponse,PayeAtsData]] = {
+  def getPayeATSData(nino: Nino, taxYear: Int)(implicit hc: HeaderCarrier, request: PayeAuthenticatedRequest[_]):Future[Either[HttpResponse,PayeAtsData]] = {
      middleConnector.connectToPayeATS(nino,taxYear) map { response =>
        response status match {
-         case OK =>  Right(response.json.as[PayeAtsData])
+         case OK =>  {
+           sendAuditEvent(taxYear)
```

Comment:
What happens if sendAuditEvent fails?

Reply:
As discussed going to log the exception and prevent user from using the service if  sendAuditEvent fails.
Need to confirm from performance analyst regarding the cache bit.

Reply:
> Also do we want to audit every time, even when we get data from cache?
This is an outstanding question, which will need to be clarified as part of the ticket. I've emailed about getting this created properly but haven't heard back yet.